aspnetcore/mvc/razor-pages/ui-class.md

```diff
@@ -0,0 +1,64 @@
+---
+title:  Reusable Razor UI in class libraries with ASP.NET Core
+author: Rick-Anderson
+description: Explains how to create reusable Razor Pages UI in a class library.
+manager: wpickett
+ms.author: riande
+ms.date: 3/31/2018
+ms.prod: asp.net-core
+ms.technology: aspnet
+ms.topic: advanced
+uid: mvc/razor-pages/ui-class
+---
+# Reusable Razor User Interface (UI) in class libraries with ASP.NET Core
+
+By [Rick Anderson](https://twitter.com/RickAndMSFT)
+
+Razor views, pages, controllers, page models, and data models can be packaged and shared in a class library. Applications can include the Razor UI class library and override the views and pages it contains.
+
+[!INCLUDE[](~/includes/2.1-required.md)]
+[!INCLUDE[](~/includes/2.1.md)]
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/mvc/razor-pages/ui-class/sample) ([how to download](xref:tutorials/index#how-to-download-a-sample))
+
+## Create a class library containing Razor UI
+
+* Create a .NET Standard class library. For example, run 
+
+    ```cli
+    dotnet new classlib -f netstandard2.0 -o RazorClassUI
+    ```
+    See [dotnet new](/dotnet/core/tools/dotnet-new) for more information.
+* Update the *.csproj* file to contain:
+
+    * The MSBuild properties `ResolvedRazorCompileToolset`, `RazorCompileOnBuild`, `IncludeContentInPack`.
```

Comment:
Ideally we want people to use the preview2 sdk where 

a) None of these properties are required
b) You get tooling for Razor.

Should we draft this for the preview2 Sdk and release the docs once it's available? The instructions are a bit different and lot more streamlined.